### PR TITLE
Add support for the new economy system

### PR DESCRIPTION
**Pull request by kirre-bylund**

### CODE REVIEW

File: Runtime/Client/LootLockerEndPoints.cs [hunk 1]
```diff
@@ -233,6 +234,25 @@ public class LootLockerEndPoints
         public static EndPointClass ComputeAndLockDropTable = new EndPointClass("v1/player/droptables/{0}/compute?asset_details={1}", LootLockerHTTPMethod.POST);
         public static EndPointClass PickDropsFromDropTable = new EndPointClass("v1/player/droptables/{0}/pick", LootLockerHTTPMethod.POST);
 
+        // Currencies
+        [Header("Currencies")]
+        public static EndPointClass listCurrencies = new EndPointClass("currency", LootLockerHTTPMethod.GET);
+        public static EndPointClass getCurrencyByCode = new EndPointClass("currency/code/{0]", LootLockerHTTPMethod.GET);
```
Comment (ladydascalie) [inline Runtime/Client/LootLockerEndPoints.cs:240]: should `{0]` not be `{0}`?

File: Runtime/Client/LootLockerEndPoints.cs [hunk 2]
```diff
@@ -233,6 +234,25 @@ public class LootLockerEndPoints
...
+        public static EndPointClass getCurrencyDenominations = new EndPointClass("currency/{0}/denominations", LootLockerHTTPMethod.GET);
+
+        // Balances
+        [Header("Balances")]
+        public static EndPointClass listBalancesInWallet = new EndPointClass("balances/wallet/{0}", LootLockerHTTPMethod.GET);
+        public static EndPointClass getWalletByWalletId = new EndPointClass("wallets/{0]", LootLockerHTTPMethod.GET);
```
Comment (ladydascalie) [inline Runtime/Client/LootLockerEndPoints.cs:246]: Same thing here. Unsure if this is on purpose

File: Runtime/Game/Requests/PurchaseRequest.cs [hunk 3]
```diff
@@ -37,6 +37,41 @@ public class LootLockerAndroidPurchaseVerificationRequest
         public int asset_id { get; set; }
         public string purchase_token { get; set; }
     }
+
+    public class LootLockerPurchaseCatalogItemResponse : LootLockerResponse
+    {
+
```
Comment (JohannesLoot) [inline Runtime/Game/Requests/PurchaseRequest.cs:43]: Is this empty on purpose? And if so, that function could use a LootLockerResponse instead?

**Review by JohannesLoot — APPROVED** [review]
Could not find anything strange apart from my comment. So will approve 👍 
Will give more feedback (probably), when the back-end is fully released.

**Comment by kirre-bylund** [comment]
Add support for create wallets internally and do that if none are found

File: Runtime/Client/LootLockerEndPoints.cs (hunk 2 above)
Comment (kirre-bylund) [reply Runtime/Client/LootLockerEndPoints.cs]: Fixed

File: Runtime/Client/LootLockerEndPoints.cs (hunk 1 above)
Comment (kirre-bylund) [reply Runtime/Client/LootLockerEndPoints.cs]: Fixed

**Pull request merged**